### PR TITLE
14213 Removed workaround + used new corpSubType (IA/reg legal type)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14213

*Description of changes:*
- updated name(), type() and typeDescription()
- updated Business and CorpType interfaces to match API response
- deleted obsolete CCC->CC code
- updated unit tests
- app version = 1.1.22

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 1]
```diff
@@ -343,7 +343,8 @@ export default class AffiliatedEntityTable extends Mixins(DateMixin) {
     if (this.isTemporaryBusiness(business)) {
       return this.tempDescription(business)
     }
-    return GetCorpFullDescription(business.corpType.code as CorpTypeCd)
+    const code: unknown = business.corpType.code
+    return GetCorpFullDescription(code as CorpTypeCd)
```
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:347]: Can't directly cast one type to another without doing this.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 2]
```diff
@@ -367,8 +368,8 @@ export default class AffiliatedEntityTable extends Mixins(DateMixin) {
     }
     // if this is an IA or registration then show legal type
     if (this.isTemporaryBusiness(business)) {
-      const legalType: unknown = business.corpType.legalType
-      return GetCorpFullDescription(legalType as CorpTypeCd)
+      const legalType: unknown = business.corpSubType?.code
+      return GetCorpFullDescription(legalType as CorpTypeCd) // may return ''
```
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:372]: Temporary business = IA or Registration

In this case, look at the new object to get the legal type.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 3]
```diff
@@ -474,10 +475,10 @@ export default class AffiliatedEntityTable extends Mixins(DateMixin) {
 
     let filingResponse = null
 
-    if (regTypes.includes(business.nameRequest?.legalType as CorpTypes)) {
```
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:477]: These are the same types -- no need to cast.

File: auth-web/src/models/business.ts [hunk 4]
```diff
@@ -14,20 +14,20 @@ export interface FolioNumberload {
...
 export interface Business {
     businessIdentifier: string
     businessNumber?: string
     name?: string
     contacts?: Contact[]
-    corpType: CorpType,
-    folioNumber: string,
-    lastModified?: string,
+    corpType: CorpType
+    corpSubType?: CorpType
```
Comment (severinbeauvais) [inline auth-web/src/models/business.ts:28]: `corpSubType` doesn't exist for all businesses.

File: auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { createLocalVue, shallowMount } from '@vue/test-utils'
+import { createLocalVue, mount } from '@vue/test-utils'
```
Comment (severinbeauvais) [inline auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts:1]: Need to deep-mount to render the Vuetify table.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 1 above)
Comment (ChrisGabel23) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:347]: How was this working before? Or are you fixing something that wasn't working before?

**Review by ChrisGabel23 — APPROVED** [review]

**Comment by seeker25** [comment]
@severinbeauvais does this need to go out ASAP?

**Review by seeker25 — APPROVED** [review]

**Review by lekshmimallika-aot — APPROVED** [review]

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 1 above)
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:347]: `business.corpType.code` was previously a string, but now it's of type CorpTypes.

**Comment by severinbeauvais** [comment]
^^ The testing-coverage step failed, but those are not new unit test failures. I created ticket #14302 to fix those, as they are not in the scope of THIS ticket.

**Comment by severinbeauvais** [comment]
> @severinbeauvais does this need to go out ASAP?

No rush. Thanks for asking :)

**Pull request merged**